src/current/_includes/v23.2/sql/insert-vs-upsert.md

```diff
@@ -1,9 +1,3 @@
-When inserting or updating all columns of a table, and the table has no secondary
-indexes, Cockroach Labs recommends using an `UPSERT` statement instead of the
-equivalent `INSERT ON CONFLICT` statement. Whereas `INSERT ON CONFLICT` always
-performs a read to determine the necessary writes, the `UPSERT` statement writes
-without reading, making it faster. This may be particularly useful if
-you are using a simple SQL table of two columns to [simulate direct KV access]({% link {{ page.version.version }}/sql-faqs.md %}#can-i-use-cockroachdb-as-a-key-value-store).
-In this case, be sure to use the `UPSERT` statement.
+When inserting or updating columns on a table that does not have [secondary indexes]({% link {{ page.version.version }}/indexes.md %}), Cockroach Labs recommends using an `UPSERT` statement instead of `INSERT ON CONFLICT DO UPDATE`. Whereas `INSERT ON CONFLICT` always performs a read, the `UPSERT` statement writes without reading, making it faster. This may be useful if you are using a simple SQL table of two columns to [simulate direct KV access]({% link {{ page.version.version }}/sql-faqs.md %}#can-i-use-cockroachdb-as-a-key-value-store).
 
-For tables with secondary indexes, there is no performance difference between `UPSERT` and `INSERT ON CONFLICT`.
+If the table has a secondary index, there is no performance difference between `UPSERT` and `INSERT ON CONFLICT`. However, `INSERT` may not scan the table for existing values. This can provide a performance improvement over `UPSERT`.
```

Comment:
nit: Maybe make it super clear that the second `INSERT` lacks `ON CONFLICT`, like:

> However, `INSERT` without an `ON CONFLICT` clause may not scan...